src/interface/web/app/search/page.tsx

```diff
@@ -153,41 +292,200 @@ function focusNote(note: SearchResult) {
+    const [isDragAndDropping, setIsDragAndDropping] = useState(false);
+
+    const [warning, setWarning] = useState<string | null>(null);
+    const [error, setError] = useState<string | null>(null);
+    const [uploading, setUploading] = useState(false);
+    const [progressValue, setProgressValue] = useState(0);
+    const fileInputRef = useRef<HTMLInputElement>(null);
+
+    useEffect(() => {
+        if (!uploading) {
+            setProgressValue(0);
+            if (!warning && !error) {
+                // Force close the dialog by simulating a click on the escape key
+                const event = new KeyboardEvent("keydown", { key: "Escape" });
+                document.dispatchEvent(event);
+            }
+        }
+
+        if (uploading) {
+            const interval = setInterval(() => {
+                setProgressValue((prev) => {
+                    const increment = Math.floor(Math.random() * 5) + 1; // Generates a random number between 1 and 5
+                    const nextValue = prev + increment;
+                    return nextValue < 100 ? nextValue : 100; // Ensures progress does not exceed 100
+                });
+            }, 800);
+            return () => clearInterval(interval);
+        }
+    }, [uploading]);
+
+    function handleDragOver(event: React.DragEvent<HTMLDivElement>) {
+        event.preventDefault();
+        setIsDragAndDropping(true);
+    }
+
+    function handleDragLeave(event: React.DragEvent<HTMLDivElement>) {
+        event.preventDefault();
+        setIsDragAndDropping(false);
+    }
+
+    function handleDragAndDropFiles(event: React.DragEvent<HTMLDivElement>) {
+        event.preventDefault();
+        setIsDragAndDropping(false);
+
+        if (!event.dataTransfer.files) return;
+
+        uploadFiles(event.dataTransfer.files);
+    }
+
+    function openFileInput() {
+        if (fileInputRef && fileInputRef.current) {
+            fileInputRef.current.click();
+        }
+    }
+
+    function handleFileChange(event: React.ChangeEvent<HTMLInputElement>) {
+        if (!event.target.files) return;
+
+        uploadFiles(event.target.files);
+    }
+
+    function uploadFiles(files: FileList) {
+        uploadDataForIndexing(files, setWarning, setUploading, setError, setUploadedFiles);
+    }
+
+    return (
+        <Dialog>
+            <DialogTrigger asChild>
+                <Button variant={"secondary"} className="mt-4">
+                    <Brain className="h-4 w-4 mr-2" />
+                    Add Documents
+                </Button>
+            </DialogTrigger>
+            <DialogContent>
+                <DialogHeader>
+                    <DialogTitle>Build Your Knowledge Base</DialogTitle>
+                    <DialogDescription>
+                        Add your files to supercharge Khoj`&apos;`s AI with your knowledge. Get
+                        instant, personalized answers powered by your own documents and data.
+                    </DialogDescription>
```

Comment:
1. Why wrap the apostrophe in backticks?
2. An alternate description for the dialog

```suggestion
                    <DialogTitle>Build Your Knowledge Base</DialogTitle>
                    <DialogDescription>
                        Add documents to your Khoj knowledge base.
                        Quickly search and get personalized answers from your documents.
                    </DialogDescription>
```

Reply:
> Why wrap the apostrophe in backticks?

It's escaping it so that it can be rendered correctly at export time.

Suggestion sounds good, will incorporate.